extensions/ql-vscode/test/vscode-tests/no-workspace/model-editor/method-modeling/method-modeling-view-provider.test.ts

```diff
@@ -0,0 +1,212 @@
+import { createMockModelingEvents } from "../../../../__mocks__/model-editor/modelingEventsMock";
+import { createMockModelingStore } from "../../../../__mocks__/model-editor/modelingStoreMock";
+import { mockedObject } from "../../../../mocked-object";
+import type { FromMethodModelingMessage } from "../../../../../src/common/interface-types";
+import { DisposableObject } from "../../../../../src/common/disposable-object";
+import type { ModelingEvents } from "../../../../../src/model-editor/modeling-events";
+import type {
+  DbModelingState,
+  SelectedMethodDetails,
+} from "../../../../../src/model-editor/modeling-store";
+import { mockDatabaseItem } from "../../../utils/mocking.helpers";
+import {
+  createMethod,
+  createUsage,
+} from "../../../../factories/model-editor/method-factories";
+
+describe("method modeling view provider", () => {
+  // Modeling store
+  let getStateForActiveDb: jest.Mock<DbModelingState | undefined, []>;
+  let getSelectedMethodDetails: jest.Mock<
+    SelectedMethodDetails | undefined,
+    []
+  >;
+
+  // Modeling events
+  let selectedMethodChangedEventEmitter: ModelingEvents["onSelectedMethodChangedEventEmitter"];
+  let dbOpenedEventEmitter: ModelingEvents["onDbOpenedEventEmitter"];
+
+  // View provider
+  let viewProvider: MethodModelingViewProvider;
+  let onDidReceiveMessage: (msg: FromMethodModelingMessage) => Promise<void>;
+  let postMessage: (message: unknown) => Promise<boolean>;
+
+  beforeEach(async () => {
+    const app = createMockApp({});
+
+    getStateForActiveDb = jest.fn().mockReturnValue(undefined);
+    getSelectedMethodDetails = jest.fn().mockReturnValue(undefined);
+    const modelingStore = createMockModelingStore({
+      getStateForActiveDb,
+      getSelectedMethodDetails,
+    });
+
+    selectedMethodChangedEventEmitter = new EventEmitter();
+    dbOpenedEventEmitter = new EventEmitter();
+    const modelingEvents = createMockModelingEvents({
+      onSelectedMethodChanged: selectedMethodChangedEventEmitter.event,
+      onDbOpened: dbOpenedEventEmitter.event,
+    });
+
+    const modelConfigListener = mockedObject<ModelConfigListener>({
+      showTypeModels: true,
+      onDidChangeConfiguration: jest.fn(),
+    });
+
+    viewProvider = new MethodModelingViewProvider(
+      app,
+      modelingStore,
+      modelingEvents,
+      modelConfigListener,
+    );
+
+    postMessage = jest.fn().mockResolvedValue(true);
+    const webview: Webview = {
+      options: {},
+      html: "",
+      onDidReceiveMessage: (listener) => {
+        onDidReceiveMessage = listener;
+        return new DisposableObject();
+      },
+      postMessage,
+      asWebviewUri: (uri: Uri) => uri,
+      cspSource: "",
+    };
+
+    const webviewView = mockedObject<WebviewView>({
+      webview,
+      onDidDispose: jest.fn(),
+    });
+
+    viewProvider.resolveWebviewView(
+      webviewView,
+      mockedObject<WebviewViewResolveContext>({}),
+      new CancellationTokenSource().token,
```

Comment:
It seems like it should be safe to remove these parameters from the `resolveWebviewView` method implementation so you don't need to pass these.

Reply:
`resolveWebviewView` is implementing a method from the `WebviewViewProvider` provider so that's why it had those unused arguments. They're there for documentation purposes, though personally I agree I think it's fine to not define them on our definition if we're not using them.

Previously it was just a matter of documentation preference with no other concerns, but now that we're calling this method ourselves in the tests it's making our testing harder as we have to mock more objects. So I think that's enough justification to remove them now.